### ISSUE TITLE
Syntax highlighting is wrong when using parentheses in comments

### ISSUE THREAD

**Issue by roryokane** [fish-shell/fish-shell#1990]
When typing a comment in Fish that includes an open parenthesis `(`, the syntax highlighting highlights the part after `(` as if it is a sub-command rather than part of the comment. For example, if you type this:

```
# a normal comment
# the old command (echo bar) didn't work
# do the thing (because it's useful) and then the other thing
```

You see this:

![demonstration of broken syntax highlighting](https://cloud.githubusercontent.com/assets/79168/6694088/11845fae-ccae-11e4-8e05-a4e8f50724ba.png)

This highlighting is confusing. It makes me worried that words in parentheses are actually going to be executed as a command, though thankfully they are not.

**Comment by roryokane** [comment]
I think the problem is somewhere [in `highlighter_t::color_argument`](https://github.com/fish-shell/fish-shell/blob/48d353610fc5fac5a9d491e9d0ad9c7e4c79fe7e/highlight.cpp#L1012) in `highlight.cpp`. That function handles the highlighting of command substitutions. It looks for sub-commands using [`parse_util_locate_cmdsubst_range`](https://github.com/fish-shell/fish-shell/blob/48d353610fc5fac5a9d491e9d0ad9c7e4c79fe7e/parse_util.cpp#L291) in `parse_util.cpp`.

**Comment by snnw** [comment]
Could you try this with the latest Git version? Either build from source our install a nightly: https://github.com/fish-shell/fish-shell/wiki/Nightly-builds .

It works as expected for me with version 2.1.2-1224-g48d3536

**Comment by xfix** [comment]
Duplicate of #1163. Closing.

**Comment by roryokane** [comment]
@snnw Thanks, highlighting works for me when I install version 2.1.2-1224-g48d3536 with `brew install fish --HEAD`.

When it didn’t work, I was using the released version 2.1.2, the latest version [in Homebrew](http://braumeister.org/formula/fish). The version I was using [is downloaded from http://fishshell.com/files/2.1.2/fish-2.1.2.tar.gz](https://github.com/Homebrew/homebrew/blob/967cb9ca97a830ce2c85b21fdbf7756629198a25/Library/Formula/fish.rb#L3). So there was a regression some time after #1163, though now it is fixed again.

**Comment by zanchey** [comment]
Sadly the fix in #1163 has not yet been included in a full release (2.1.2 is built from the 2.1 branch).
